swupd/archive_test.go

```diff
@@ -0,0 +1,115 @@
+package swupd
+
+import (
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+func TestNewCompressedTarReaderUncompressedRegularFile(t *testing.T) {
+	testCases := []string{
+		"/etc/protocols",
+		"/usr/share/defaults/etc/protocols",
+		"/usr/share/doc/systemd/LICENSE.GPL2",
+	}
+	for _, tc := range testCases {
+		var f *os.File
+		var err error
+		if f, err = os.Open(tc); err != nil {
+			// not checking non-existent files
+			// not testing the os.Open call
+			continue
+		}
+		defer func() {
+			_ = f.Close()
+		}()
+		_, err = NewCompressedTarReader(f)
```

Comment:
I realize now this is a not good name to really explain that this is actually opening and uncompressing a file from <whatever> compression type. :+1: though